### ISSUE TITLE
Feature request: remove dead `has` branches at source->source optimization step

### ISSUE THREAD

**Issue by jm355** [markw65/prettier-extension-monkeyc#25]
sdk 7 brought the ability to compile out dead branches based on has checks, however [they're not enabled by default and most devices won't get this until new device configs get rolled out](https://forums.garmin.com/developer/connect-iq/f/discussion/397832/if-sdk-7-can-compile-out-unused-branches-with-the-has-check-why-are-binaries-with-it-still-consistently-larger). Can prettier monkeyc do this at the source->source optimization step?

**Comment by markw65** [comment]
Yes - I think I need to do some of it under an option, because people have argued that they don't want `Foo has :Bar` to be optimized to false for a device, since it's possible that a newer version of the firmware will support that feature, and they don't want to have to keep updating their builds.

On the other hand it's always safe to optimize a condition to true, when the device is guaranteed to support it, because th .iq file won't install on devices older than the one you build against.

**Comment by markw65** [comment]
If you check for something that isn't defined anywhere (including in the `Toybox`), it will optimize the `has` check to false, because there's no possibility of any other result.

For things that it does find, there are two possibilities:
 - it's defined by the system, in which case whether it's actually available or not may depend on the device you're compiling for
 - it's defined by your code, in which case it definitely exists (assuming we've already dealt with exclude annotations).

I currently don't distinguish between those cases, so I never optimize `has` checks for things that have a definition.

My proposed change is:
 - always optimize checks of user defined symbols to `true`
 - always optimize checks of `Toybox` symbols that are available to the current device to `true`
 - under a new option, optimize checks of `Toybox` symbols that are not available to the current device to `false`

**Comment by jm355** [comment]
Sounds good to me!
